lib/usd/ui/layerEditor/layerTreeModel.cpp

```diff
@@ -51,6 +51,68 @@ namespace {
 const QString LAYER_EDITOR_MIME_TYPE = QStringLiteral("text/plain");
 const QString LAYED_EDITOR_MIME_SEP = QStringLiteral(";");
 
+bool isPathInside(const std::string& parentDir, const std::string& childPath)
+{
+    fs::path parent = fs::weakly_canonical(parentDir);
+    fs::path child = fs::weakly_canonical(childPath);
+
+    // Iterate up from child to root
+    for (fs::path p = child; !p.empty(); p = p.parent_path()) {
+        if (p == parent)
+            return true;
+
+        fs::path next = p.parent_path();
+        if (next == p) // reached root (ex "C:\")
+            break;
+    }
+    return false;
+}
+
+bool shouldDisplayComponentInitialSaveDialog(
+    const UsdStageRefPtr stage,
+    const std::string&   proxyShapePath)
+{
+    MString defineIsComponentCmd;
+    defineIsComponentCmd.format(
+        "def usd_component_creator_is_proxy_shape_a_component():\n"
+        "    from pxr import Sdf, Usd, UsdUtils\n"
+        "    import mayaUsd\n"
+        "    import mayaUsd.ufe\n"
+        "    try:\n"
+        "        from AdskUsdComponentCreator import ComponentDescription\n"
+        "    except ImportError:\n"
+        "        return -1\n"
+        "    proxyStage = mayaUsd.ufe.getStage(\"^1s\")\n"
+        "    component_description = ComponentDescription.CreateFromStageMetadata(proxyStage)\n"
+        "    if component_description:\n"
+        "        return 1\n"
+        "    else:\n"
+        "        return 0",
+        proxyShapePath.c_str());
+
+    int isStageAComponent = 0;
+    if (MS::kSuccess == MGlobal::executePythonCommand(defineIsComponentCmd, false, false)) {
+        MString runIsComponentCmd = "usd_component_creator_is_proxy_shape_a_component()";
+        MGlobal::executePythonCommand(runIsComponentCmd, isStageAComponent);
+    }
+
+    // Failed to import the component creator python libraries
+    if (isStageAComponent == -1) {
+        return false;
+    }
+
+    if (isStageAComponent == 0) {
+        TF_WARN("Stage at proxy path '%s' is not a component.", proxyShapePath.c_str());
```

Comment:
I don't think not being a component is an error? Normal stages need saving too. I would only issue an error if the call to MGlobal::execute fails. (i.e an else at line 97)